kyaml/kio/filters/fmtr_test.go

```diff
@@ -904,3 +905,73 @@ func TestFormatFileOrDirectory_trimWhiteSpace(t *testing.T) {
 
 	assert.Equal(t, string(testyaml.FormattedYaml1), string(b))
 }
+
+func TestFormatFileOrDirectory_FmtAnnotation(t *testing.T) {
+	testCases := []struct {
+		name               string
+		fmtAnnotationValue string
+		expectFormatting   bool
+		expectError        bool
+	}{
+		{
+			name:               "no formatting annotation",
+			fmtAnnotationValue: "",
+			expectFormatting:   true,
+		},
+		{
+			name:               "formatting strategy none",
+			fmtAnnotationValue: FmtStrategyNone,
+			expectFormatting:   false,
+		},
+		{
+			name:               "formatting strategy standard",
+			fmtAnnotationValue: FmtStrategyStandard,
+			expectFormatting:   true,
+		},
+		{
+			name:               "unknown formatting strategy",
+			fmtAnnotationValue: "unknown",
+			expectError:        true,
+		},
+	}
+
+	for i := range testCases {
+		test := testCases[i]
+		t.Run(test.name, func(t *testing.T) {
+			f, err := ioutil.TempFile("", "yamlfmt*.yaml")
+			assert.NoError(t, err)
+			defer os.Remove(f.Name())
```

Comment:
Is there enough infra in place to do these tests without going out to disk?

Reply:
Most tests in kyaml uses this pattern I think. So if we want to avoid writing to disk, we should find a way to do it and use it consistently across all the tests.

Reply:
agreed.  We've got a fake FS in the kustomize API we could switch over to someday, modulo circular dep problems :P